samples/features/azure-arc-data-controller/deployment/kubeadm/ubuntu-single-node-vm/setup-controller.sh

```diff
@@ -0,0 +1,299 @@
+
+swapoff -a
+sed -i '/swap/s/^\(.*\)$/#\1/g' /etc/fstab
+
+curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+
+cat <<EOF >/etc/apt/sources.list.d/kubernetes.list
+
+deb http://apt.kubernetes.io/ kubernetes-xenial main
+
+EOF
+
+# Install docker and packages to allow apt to use a repository over HTTPS.
+#
+apt-get update -q
+
+apt-get install -q -y ebtables ethtool
+
+#apt-get install -y docker.ce
+
+apt-get install -q -y apt-transport-https
+
+# Setup daemon.
+#
+cat > /etc/docker/daemon.json <<EOF
+{
+  "exec-opts": ["native.cgroupdriver=systemd"],
+  "log-driver": "json-file",
+  "log-opts": {
+    "max-size": "100m"
+  },
+  "storage-driver": "overlay2"
+}
+EOF
+
+mkdir -p /etc/systemd/system/docker.service.d
+
+# Restart docker.
+#
+systemctl daemon-reload
+systemctl restart docker
+
+apt-get install -q -y kubelet=$KUBE_DPKG_VERSION kubeadm=$KUBE_DPKG_VERSION kubectl=$KUBE_DPKG_VERSION
+
+# Holding the version of kube packages.
+#
+apt-mark hold kubelet kubeadm kubectl
+curl https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get | bash
+
+. /etc/os-release
+if [ "$UBUNTU_CODENAME" == "bionic" ]; then
+    modprobe br_netfilter
+fi
+
+# Disable Ipv6 for cluster endpoints.
+#
+sudo sysctl -w net.ipv6.conf.all.disable_ipv6=1
+sudo sysctl -w net.ipv6.conf.default.disable_ipv6=1
+sudo sysctl -w net.ipv6.conf.lo.disable_ipv6=1
+
+echo net.ipv6.conf.all.disable_ipv6=1 > /etc/sysctl.conf
+echo net.ipv6.conf.default.disable_ipv6=1 > /etc/sysctl.conf
+echo net.ipv6.conf.lo.disable_ipv6=1 > /etc/sysctl.conf
+
+
+sysctl net.bridge.bridge-nf-call-iptables=1
+
+# Setting up the persistent volumes for the kubernetes.
+#
+for i in $(seq 1 $PV_COUNT); do
+
+  vol="vol$i"
+
+  mkdir -p /mnt/local-storage/$vol
+
+  mount --bind /mnt/local-storage/$vol /mnt/local-storage/$vol
+
+done
+echo "Kubernetes pre-requisites have been completed." 
+
+# Setup kubernetes cluster including remove taint on master.
+#
+echo ""
+echo "#############################################################################"
+echo "Starting to setup Kubernetes master..." 
+
+# Initialize a kubernetes cluster on the current node.
+#
+sudo kubeadm init --pod-network-cidr=10.244.0.0/16 --kubernetes-version=$KUBE_VERSION
+
+mkdir -p $HOME/.kube
+mkdir -p /home/$SUDO_USER/.kube
+
+sudo cp -f /etc/kubernetes/admin.conf $HOME/.kube/config
+sudo chown $(id -u $SUDO_USER):$(id -g $SUDO_USER) $HOME/.kube/config
+
+# To enable a single node cluster remove the taint that limits the first node to master only service.
+#
+master_node=`kubectl get nodes --no-headers=true --output=custom-columns=NAME:.metadata.name`
+kubectl taint nodes ${master_node} node-role.kubernetes.io/master:NoSchedule-
+
+# Local storage provisioning.
+#
+kubectl apply -f https://raw.githubusercontent.com/ananto-msft/sql-server-samples/$GITHUB_AADATACONTROLLER_BRANCH/samples/features/azure-arc-data-controller/deployment/kubeadm/ubuntu/local-storage-provisioner.yaml
```

Comment:
Update to actual link based on the suggested folder structure.

Reply:
Done.